pkg/await/apps_deployment.go

```diff
@@ -394,6 +395,23 @@ func (dia *deploymentInitAwaiter) processDeploymentEvent(event watch.Event) {
 		// No current generation, Deployment controller has not yet created a ReplicaSet. Do
 		// nothing.
 		return
+	} else {
+		if currentGenerationInt, err := strconv.Atoi(dia.currentGeneration); err == nil {
```

Comment:
Instead of an `else` it seems like this should be hidden behind an `if extensionsv1beta1API {`. In particular, it seems like we _don't_ want to affect the code paths of the other versions, since they already work.

Reply:
I was thinking that these conditions applied generally, but I think you're right. It was working well before, so I'm going to make this conditional on the older API version.